superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx

```diff
@@ -54,21 +73,86 @@ const FilterControls: FC<FilterControlsProps> = ({
+  if (!lastFocusedTabId || !dashboardHasTabs) {
+    filtersInScope = cascadeFilters;
+  } else {
+    cascadeFilters.forEach((filter, index) => {
+      if (cascadeFilters[index].tabsInScope?.includes(lastFocusedTabId)) {
+        filtersInScope.push(filter);
+      } else {
+        filtersOutOfScope.push(filter);
+      }
+    });
+  }
+
   return (
     <Wrapper>
-      {cascadeFilters.map(filter => (
-        <CascadePopover
-          data-test="cascade-filters-control"
-          key={filter.id}
-          visible={visiblePopoverId === filter.id}
-          onVisibleChange={visible =>
-            setVisiblePopoverId(visible ? filter.id : null)
-          }
-          filter={filter}
-          onFilterSelectionChange={onFilterSelectionChange}
-          directPathToChild={directPathToChild}
-        />
+      {portalNodes.map((node, index) => (
+        <portals.InPortal node={node}>
+          <CascadePopover
+            data-test="cascade-filters-control"
+            key={cascadeFilters[index].id}
+            visible={visiblePopoverId === cascadeFilters[index].id}
+            onVisibleChange={visible =>
+              setVisiblePopoverId(visible ? cascadeFilters[index].id : null)
+            }
+            filter={cascadeFilters[index]}
+            onFilterSelectionChange={onFilterSelectionChange}
+            directPathToChild={directPathToChild}
+          />
+        </portals.InPortal>
       ))}
+      {filtersInScope.map(filter => {
+        const index = cascadeFilters.findIndex(f => f.id === filter.id);
+        return <portals.OutPortal node={portalNodes[index]} />;
+      })}
+      {showCollapsePanel && (
+        <Collapse
+          ghost
+          bordered
+          expandIconPosition="right"
+          collapsible={filtersOutOfScope.length === 0 ? 'disabled' : undefined}
+          css={theme => css`
+            &.ant-collapse {
+              margin-top: ${filtersInScope.length > 0
+                ? theme.gridUnit * 6
+                : 0}px;
+              & > .ant-collapse-item {
+                & > .ant-collapse-header {
+                  padding-left: 0;
+                  padding-bottom: ${theme.gridUnit * 2}px;
+
+                  & > .ant-collapse-arrow {
+                    right: ${theme.gridUnit}px;
+                  }
+                }
+
+                & .ant-collapse-content-box {
+                  padding: ${theme.gridUnit * 4}px 0 0;
+                }
+              }
+            }
+          `}
+        >
+          <Collapse.Panel
+            header={`${t('Filters out of scope')} (${
+              filtersOutOfScope.length
+            })`}
+            key="1"
+          >
+            {filtersOutOfScope.map(filter => {
+              const index = cascadeFilters.findIndex(f => f.id === filter.id);
+              return <portals.OutPortal node={portalNodes[index]} />;
+            })}
+          </Collapse.Panel>
+        </Collapse>
```

Comment:
I wonder if it would be possible to postpone rendering of the out of scope components to the time they become visible for the first time, similar to how charts in tabs only load up when switching tabs? Especially with dashboards with lots of filters and tabs that only have a few filters in scope per tab, this could have a performance boost as we would only need to render the filters as they are needed.

I checked that it's possible to pass default props in the `InPortal` and can then be changed/added to in the `OutPortal`, but it seems this may add quite a bit of additional logic that might obfuscate the original code, so this additional feature probably shouldn't be in scope for this PR..

Reply:
Hmm, besides more convoluted logic, I wonder if it would be confusing for the user. Imagine that you add a filter to a dashboard scoped to 1 tab, you come back to that dashboard later and your filter is nowhere to be found, until you click on that tab. I can see the potential benefits and I'm all for implementing that, but I think we should think this through first so that it not only performs optimally, but also looks good 🙂

Reply:
I think you misunderstood what I was getting at. What I meant was that as long as the "Out of scope" collapsible is collapsed, the requests for those filters would not go out. So in this case:
![image](https://user-images.githubusercontent.com/33317356/120469829-89950200-c3ab-11eb-93db-ef44dbcc0a91.png)

the request for that last filter would not yet have gone out, but once it's expanded

![image](https://user-images.githubusercontent.com/33317356/120469915-a16c8600-c3ab-11eb-92ba-bf37d4cc2452.png)

The charts would render for the first time, triggering the requests. If the user moved to one of the tabs that are in scope for that filter, it would also trigger that query, but of course only once (the next time it goes out and in scope, it would already have been loaded hence no need to retrigger the query).

But yes, this was more a thought for future improvement, no need to delay this PR with this now.

Reply:
Ahh, I understand now, thanks for explanation. Let's implement it in the next iteration!